src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.MoveType
+{
+    public partial class MoveTypeTests : CSharpMoveTypeTestsBase
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveTypeWithNoContainerNamespace()
+        {
+            var code = 
+@"[||]class Class1 { }
+class Class2 { }";
+            var codeAfterMove = @"class Class2 { }";
+
+            var expectedDocumentName = "Class1.cs";
+            var destinationDocumentText = @"class Class1 { }";
+
+            await TestMoveTypeToNewFileAsync(code, codeAfterMove, expectedDocumentName, destinationDocumentText);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveTypeWithWithUsingsAndNoContainerNamespace()
+        {
+            var code =
+@"// Banner Text
+using System;
+
+[||]class Class1 { }
+class Class2 { }";
+
+            var codeAfterMove =
+@"// Banner Text
+
+class Class2 { }";
+
+            var expectedDocumentName = "Class1.cs";
+            var destinationDocumentText = 
+@"class Class1 { }";
```

Comment:
I think I've answered it in answering another question about how we decide what to keep and what to lose.